dired-filter.el

```diff
@@ -642,20 +642,35 @@ The matched lines are returned as a string."
     map)
   "Keymap used when over a group header.")
 
-(defun dired-filter-group--make-header (name &optional collapsed)
-  "Make group header named by NAME.
+(defun dired-filter-group--insert-header (name &optional collapsed)
+  "Insert a group header named by NAME.
 
 Optional argument COLLAPSED specifies if the header is collapsed
 by default."
-  (concat (propertize
-           (concat
-            "  "
-            (propertize (format "[ %s%s ]" name (if collapsed " ..." ""))
-                        'font-lock-face 'dired-filter-group-header))
-           'keymap dired-filter-group-header-map
-           'dired-filter-group-collapsed collapsed
-           'dired-filter-group-header name)
-          "\n"))
+  (insert
```

Comment:
I moved the insert here because of the dependency on `point' and since `dired-filter-group--make-header` was only ever called when the header was going to be inserted. But I see your point of having pure functions instead of functions which produce side effects.